rafthttp/http.go

```diff
@@ -91,7 +91,7 @@ func (h *pipelineHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	if from, err := types.IDFromString(r.Header.Get("X-Server-From")); err != nil {
+	if from, err := types.IDFromString(r.Header.Get("X-Server-From")); err == nil {
```

Comment:
suggest to use one new func to implement this duplicate code, as they are the same logic but you must change it two places.